cmake/linux/helpers.cmake

```diff
@@ -59,8 +59,8 @@ function(set_target_properties_obs target)
     set_target_properties(
       ${target}
       PROPERTIES
-        VERSION ${OBS_VERSION_CANONICAL}
-        SOVERSION ${OBS_VERSION_MAJOR}
+        VERSION 30
```

Comment:
Is this change by design?

Reply:
Setting `VERSION` to the same value as `SOVERSION` will make CMake generate only one file (here `.30`) without symlink.

Libraries link to libobs (so plugins) will always search for a matching library marked with 30 as SOVERSION.
Not generating `.30.y.z` should cause no issue.

This is by design, until the ABI/API version management of libobs (and obs-frontend) is decoupled from OBS Studio.

Reply:
If I understood it correctly, then the main idea is that the actual file shipped is always tied to the release version (e.g. `libobs.so.31.2.0`) and that `SOVERSION` is used to generate an ABI-version symlink (e.g. `libobs.so.30`).

I'm not particularly concerned about breaking this pattern, as long as we do it consciously.

Reply:
You got what `VERSION` and `SOVERSION` set to CMake.
Normally it should something like `libobs.so.30.1.2` and `libobs.so.30` since the former matches the latter. Mismatching major number is not good in library filename is not appropriate.

So yes, this PR consciously "disable" the presence of a `.so` file with the full version to only have the `.30`.

Once we decouple the versioning of OBS Studio and libobs, the full version variant can be restored.

Reply:
Is this a blocking concern, or is this line of questioning just for clarification?

Reply:
Just clarification, no issues.